modules/actor/actor-wfrp4e.js

```diff
@@ -1283,125 +1283,155 @@ export default class ActorWfrp4e extends WFRP4eDocumentMixin(Actor)
     }
   }
 
+  /**
+   * Decrements all injuries of this Actor
+   *
+   * @return {Promise<void>}
+   */
   async decrementInjuries() {
-    this.injuries.forEach(i => this.decrementInjury(i))
+    for (let i of this.injuries)
+      await this.decrementInjury(i);
   }
 
+  /**
+   * Decrements a `duration` value for the Injury.
+   * If `duration` reaches 0, inform about Injury being finished.
+   *
+   * @param {ItemWfrp4e}injury
+   *
+   * @return {Promise<*>}
+   */
   async decrementInjury(injury) {
     if (isNaN(injury.system.duration.value))
       return ui.notifications.notify(game.i18n.format("CHAT.InjuryError", { injury: injury.name }))
 
-    injury = duplicate(injury)
-    injury.system.duration.value--
+    injury = foundry.utils.duplicate(injury);
+    injury.system.duration.value--;
 
-    if (injury.system.duration.value < 0)
+    if (injury.system.duration.value <= 0) {
       injury.system.duration.value = 0;
-
-    if (injury.system.duration.value == 0) {
-      let chatData = game.wfrp4e.utility.chatDataSetup(game.i18n.format("CHAT.InjuryFinish", { injury: injury.name }), "gmroll")
-      chatData.speaker = { alias: this.name }
-      ChatMessage.create(chatData)
+      let chatData = game.wfrp4e.utility.chatDataSetup(game.i18n.format("CHAT.InjuryFinish", { injury: injury.name }), "gmroll");
+      chatData.speaker = { alias: this.name };
+      ChatMessage.create(chatData);
     }
-    this.updateEmbeddedDocuments("Item", [injury]);
-  }
 
+    await this.updateEmbeddedDocuments("Item", [injury]);
+  }
 
+  /**
+   * Decrements all diseases of this Actor
+   *
+   * @return {Promise<void>}
+   */
   async decrementDiseases() {
-    this.diseases.forEach(d => this.decrementDisease(d))
+    for (let d of this.diseases)
+      await this.decrementDisease(d);
   }
 
+  /**
+   * Decrements a value for Disease's `incubation` or `duration` attribute, depending on whether it is active or not.
+   * If value reaches 0, activates (if `incubation`), or finishes (if `duration`) the Disease.
+   *
+   * @param {ItemWfrp4e} disease
+   *
+   * @return {Promise<void>}
+   */
   async decrementDisease(disease) {
-    let d = duplicate(disease)
-    if (!d.system.duration.active) {
-      if (Number.isNumeric(d.system.incubation.value)) {
-
-        d.system.incubation.value--
-        if (d.system.incubation.value <= 0) {
-          this.activateDisease(d)
-          d.system.incubation.value = 0;
-        }
-      }
-      else {
-        let chatData = game.wfrp4e.utility.chatDataSetup(`Attempted to decrement ${d.name} incubation but value is non-numeric`, "gmroll", false)
-        chatData.speaker = { alias: this.name }
-        ChatMessage.create(chatData)
-      }
-    }
-    else {
-      if (Number.isNumeric(d.system.duration.value)) {
+    let d = foundry.utils.duplicate(disease);
+    let type = d.system.duration.active ? 'duration' : 'incubation';
 
```

Comment:
Simplified an `if` statement that practically reused the same code in both `if` and `else` parts.